### PR TITLE
updated freeze cache versions

### PR DESCRIPTION
**Pull request by andreymheilasystems**

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -12,13 +12,14 @@ type SessionResolverBuilder interface {
 
 type SessionResolver interface {
 	SetDataFromSessionToContext(c context.Context) (context.Context, error)
-	SetDataToContext(session Session, c context.Context) (context.Context, error)
```
Comment (guybrand) [inline session.go:15]: why did you change this ?

File: sessionresolver/repos/cache/cacheRepo.go [hunk 2]
```diff
@@ -35,24 +36,27 @@ func (r cacheRepo) InsertOrUpdate(ctx context.Context, id string, obj interface{
...
 
-	for i := range versions {
+	for _, cacheCollection := range cacheCollections {
+		if filterAction != "" && !slices.IsStrExist(cacheCollection.LockVersionUpon, filterAction) {
+			continue
+		}
 		var latestVersion cacheStorage.Version
-		for _, v := range versions[i].Versions {
+		for _, v := range cacheCollection.Versions {
 			if (latestVersion.TimedTo.IsZero() || v.TimedTo.After(latestVersion.TimedTo)) && v.TimedTo.Before(now) {
```
Comment (guybrand) [inline sessionresolver/repos/cache/cacheRepo.go:67]: latestVersion.TimedTo.IsZero()  ?
where is the expiry after end of shift ?

File: sessionresolver/wrapper.go [hunk 3]
```diff
@@ -221,10 +221,26 @@ func (sw sessionWrapper) SaveSession(c context.Context, cSession session.Session
...
 
-func (sw sessionWrapper) FreezeCacheVersionsForSession(c context.Context, curSession session.Session) error {
+func (sw sessionWrapper) UnFreezeCacheVersionsForSession(c context.Context, curSession session.Session, action string) error {
 	versions := make(map[string]string)
 	fixedVersions := curSession.GetFixedCacheVersions()
-	versionsForDate, err := sw.repo.GetCacheVersions(c, curSession.GetNow())
+
+	for collection, ver := range fixedVersions {
+		versions[collection] = ver
+	}
```
Comment (guybrand) [inline sessionresolver/wrapper.go:232]: this is unfreeze, shouldnt it:
for ...
  if  !slices.IsStrExist(cacheCollection.LockVersionUpon, filterAction){
     curSession.Delete(cacheCollection)
  }
}

**Review by guybrand — CHANGES_REQUESTED** [review]

File: session.go (hunk 1 above)
Comment (andreymheilasystems) [reply session.go:15]: renamed to SetDataFromCurrentSessionToContext. because we have SetDataFromSessionToContext. and they do the same

File: sessionresolver/repos/cache/cacheRepo.go (hunk 2 above)
Comment (andreymheilasystems) [reply sessionresolver/repos/cache/cacheRepo.go:67]: latestVersion.TimedTo.IsZero() means latestVersion is empty

image

File: session.go (hunk 1 above)
Comment (orchestd) [reply session.go:15]: We're in session "domain"
code calling this would look like this:
seesion.SetDataToContext
which makes more sense than 
seesion.SetDataFromCurrentSessionToContext

SO...perhaps the change is the other way around
SetDataFromSessionToContext => SetDataToContext

**Review by orchestd — APPROVED** [review]
